src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -552,7 +552,7 @@ fn output_err_info(cmd: &ProcessBuilder, stdout: &str, stderr: &str) -> String {
 ///
 /// Note that if a `target` is specified, no args will be passed to host code (plugins, build
 /// scripts, ...), even if it is the same as the target.
-fn env_args(
+pub fn env_args(
```

Comment:
Is it perhaps possible to use `TargetInfo` directly rather than this low-level function? That structure handles things like `cfg`-related configuration as well.

Reply:
I have made this change, but doing so does appear to slow down printing with multiple `--target` options. I think this is because `TargetInfo::new` does invoke multiple process calls to rustc and it is executing inside the loop for each compile kind/target. I am also guessing that these commands are not cached because a different target is being used on each loop. It is not a huge delay, but noticeable.